.github/workflows/manifest-diff-to-pr-comment.yaml

```diff
@@ -0,0 +1,138 @@
+        description: "Which git ref to compare against (default: origin/main)"
+        default: origin/main
+
+jobs:
+  diff-file:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pull-requests: write
+
+    steps:
+      - name: Checkout current branch
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.ref }}
+          submodules: true
+          persist-credentials: false
+
+      - name: Install `skipctl` @ ${{ env.SKIPCTL_VERSION }}
+        uses: jaxxstorm/action-install-gh-release@6096f2a2bbfee498ced520b6922ac2c06e990ed2 # v2.1.0
+        with:
+          repo: kartverket/skipctl
+          tag: ${{ env.SKIPCTL_VERSION }}
+          cache: enable
+        env:
+          SKIPCTL_VERSION: ${{ inputs.skipctl-version }}
+
+      - name: Run skipctl manifests diff
+        shell: bash
+        run: |
+          set -euo pipefail
+
+          # Function to generate diff output with individual file toggles
+          generate_diff_output() {
+              local pattern=$1
+              local output=""
+
+              # First find all directories matching the pattern, then find files in them
+              while IFS= read -r -d '' dir; do
+                  while IFS= read -r -d '' file; do
+                      local diff_result
+                      diff_result=$(skipctl manifests diff -p "$file" --ref ${{ env.INPUTS_REF }} --diff-format patch 2>/dev/null || true)
+
+                      if [ -n "$diff_result" ]; then
+                          local file_path="$file"
+                          # Remove the input path prefix for cleaner display
+                          local display_path="${file_path#${{ env.INPUTS_PATH }}/}"
+
+                          output+="<details>"
+                          output+=$'\n'
+                          output+="  <summary><b>$display_path DIFF</b></summary>"
+                          output+=$'\n'
+                          output+=$'\n'
+                          output+='```diff'
+                          output+=$'\n'
+                          output+="$diff_result"
+                          output+=$'\n'
+                          output+='```'
+                          output+=$'\n'
+                          output+="</details>"
+                          output+=$'\n'
+                      fi
+                  done < <(find "$dir" -type f -print0 | sort -z)
+              done < <(find ${{ env.INPUTS_PATH }} -type d -name "$pattern" -print0 | sort -z)
+
+              echo "$output"
+          }
+
+          DIFF_OUTPUT_PROD="$(generate_diff_output '*-prod')"
+          DIFF_OUTPUT_DEV="$(generate_diff_output '*-dev')"
+
+          {
+              echo 'DIFF_OUTPUT_PROD<<EOF'
+              printf '%s\n' "$DIFF_OUTPUT_PROD"
+              echo 'EOF'
+              echo 'DIFF_OUTPUT_DEV<<EOF'
+              printf '%s\n' "$DIFF_OUTPUT_DEV"
+              echo 'EOF'
+          } >> "$GITHUB_ENV"
```

Comment:
This is not trivial to test and edit in the future 😅 

At the same time: Splitting out as a separate script is not as easy with GitHub Workflows, since the script is not automatically available 🤔 

It might be the best option, but out of curiosity: Did you consider alternatives?

Reply:
Looks good! Reviewed the other PR too.